### PR TITLE
Avoid generating parameter names that are Rust keywords

### PR DESCRIPTION
**Pull request by KiChjang**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10674)

### CODE REVIEW

**Comment by amarant** [comment]
You should add "continue" because it is a [indexeddb method](https://dxr.mozilla.org/mozilla-central/rev/d53a52b39a95dced722cca90ac74529b66dd5253/dom/webidl/IDBCursor.webidl#36).

**Comment by Manishearth** [comment]
We capitalize all DOM things so that shouldn't be an issue. However, I do agree that we should add more keywords here.

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -36,6 +36,8 @@
 CONSTRUCT_HOOK_NAME = '_constructor'
 HASINSTANCE_HOOK_NAME = '_hasInstance'
 
+RUST_KEYWORDS = {"continue", "enum", "fn", "impl", "loop", "pub", "ref", "struct", "type"}
```
Comment (Manishearth) [inline components/script/dom/bindings/codegen/CodegenRust.py]: match, etc.

https://doc.rust-lang.org/grammar.html#keywords

Comment (KiChjang) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Yup, on it.

**Comment by KiChjang** [comment]
Added all of the Rust keywords from the master list [here](https://github.com/rust-lang/rust/blob/master/src/libsyntax/parse/token.rs#L554-L608).

**Pull request merged**